django/utils/dateformat.py

```diff
@@ -30,11 +30,17 @@
 
 
 class Formatter(object):
+
+    def format_piece(self, piece):
+        if type(self.data) is datetime.date and hasattr(TimeFormat, piece):
+            raise TypeError('The format for date objects may not contain time-related format specifiers')
```

Comment:
It's checking one specifier at a time, so only one can be included. 
Would this be OK? 
`raise TypeError('The format for date objects may not contain time-related (i.e. "%s") format specifier.' % piece)`

So in case of "a" time specifier error message would look like:
TypeError: The format for date objects may not contain time-related (i.e. "a") format specifier.